File: 10.c
void swap(int *a, int *b) {
  int t = *a;
  *a = *b;
  *b = t;
}

void sortColors(int *n, int s) {
  ssize_t l = 0, m = 0, h = s - 1;
  while (m <= h) {
    if (n[m] == 0) {
      swap(&n[l++], &n[m++]);
    } else if (n[m] == 1) {
      m++;
    } else {
      swap(&n[m], &n[h--]);
    }
  }
}

File: 10.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let mut l: usize = 0;
        let mut m: usize = 0;
        let mut h: isize = nums.len() as isize - 1;

        while m as isize <= h {
            match nums[m] {
                0 => {
                    nums.swap(l, m);
                    l += 1;
                    m += 1;
                }
                1 => m += 1,
                2 => {
                    nums.swap(m, h as usize);
                    h -= 1;
                }
                _ => panic!("wrong flag"),
            }
        }
    }
}
